src/go/rpk/pkg/kafka/client_franz.go

```diff
@@ -93,17 +95,27 @@ func NewFranzClient(fs afero.Fs, p *config.RpkProfile, extraOpts ...kgo.Opt) (*k
 	}
 
 	if k.SASL != nil {
-		mech := scram.Auth{
-			User: k.SASL.User,
-			Pass: k.SASL.Password,
-		}
-		switch name := strings.ToUpper(k.SASL.Mechanism); name {
-		case "SCRAM-SHA-256", "": // we default to SCRAM-SHA-256 -- people commonly specify user & pass without --sasl-mechanism
-			opts = append(opts, kgo.SASL(mech.AsSha256Mechanism()))
-		case "SCRAM-SHA-512":
-			opts = append(opts, kgo.SASL(mech.AsSha512Mechanism()))
-		default:
-			return nil, fmt.Errorf("unknown SASL mechanism %q, supported: [SCRAM-SHA-256, SCRAM-SHA-512]", name)
+		if k.SASL.Mechanism == admin.CloudOIDC {
+			a := p.CurrentAuth()
```

Comment:
why the choice to make this a single letter...

Reply:
imo single letters are fine for this so long as we're consistent in the codebase -- I think I introduced `p` for profile and `a` for auth in my PRs that introduced profile / cloud auth. Basically the more universal something is, the shorter but also more consistent it needs to be

Reply:
@twmb  https://github.com/redpanda-data/redpanda/pull/11394#discussion_r1228763277

Reply:
Actually, `a` there as well.